src/Compilers/CSharp/Portable/Operations/CSharpOperationNodes.cs

```diff
@@ -1489,32 +1490,28 @@ protected override IOperation CreateValue()
     internal sealed partial class CSharpLazyRecursivePatternOperation : LazyRecursivePatternOperation
     {
         private readonly CSharpOperationFactory _operationFactory;
-        private readonly ImmutableArray<BoundSubpattern> _deconstructionSubpatterns;
-        private readonly ImmutableArray<BoundSubpattern> _propertySubpatterns;
-        public CSharpLazyRecursivePatternOperation(
-            CSharpOperationFactory operationFactory,
-            ITypeSymbol inputType,
-            ITypeSymbol matchedType,
-            ISymbol deconstructSymbol,
-            ImmutableArray<BoundSubpattern> deconstructionSubpatterns,
-            ImmutableArray<BoundSubpattern> propertySubpatterns,
-            ISymbol declaredSymbol,
-            SemanticModel semanticModel,
-            SyntaxNode syntax,
-            bool isImplicit)
-            : base(inputType, matchedType, deconstructSymbol, declaredSymbol, semanticModel, syntax, isImplicit)
+        private readonly BoundRecursivePattern _boundRecursivePattern;
+
+        public CSharpLazyRecursivePatternOperation(CSharpOperationFactory operationFactory, BoundRecursivePattern boundRecursivePattern, SemanticModel semanticModel)
+            : base(inputType: boundRecursivePattern.InputType,
+                   matchedType: boundRecursivePattern.DeclaredType?.Type ?? boundRecursivePattern.InputType.StrippedType(),
```

Comment:
>StrippedType [](start = 108, length = 12)

Recursive patterns are guaranteed to never match nullable types?

Reply:
Correct.  If a type is specified, it must not be nullable.  If it is not specified, it is the input type stripped of nullableness.

---
In reply to: [244242706](https://github.com/dotnet/roslyn/pull/32018#discussion_r244242706) [](ancestors = 244242706)